tests/integration/tier0/non-destructive/enabled-repositories-after-analysis/test_enabled_repositories_after_analysis.py

```diff
@@ -0,0 +1,49 @@
+import pytest
+
+from conftest import TEST_VARS
+
+
+def collect_enabled_repositories(shell):
+    """
+    Collect the enabled repositories through a subscription-manager call.
+
+    This function will return a list of repositories enabled.
+    """
+    raw_output = shell("subscription-manager repos --list-enabled").output
+    assert raw_output
+
+    enabled_repositories = []
+    for line in raw_output.splitlines():
+        if line.startswith("Repo ID:"):
+            # Get the repo_id as in that split it will be the last thing in the
+            # array.
+            repo_id = line.split("Repo ID:")[-1]
+            enabled_repositories.append(repo_id)
+
+    return enabled_repositories
+
+
+@pytest.mark.test_enabled_repositories_after_analysis
+def test_enabled_repositories_after_analysis(shell, convert2rhel, satellite_registration):
+    """Test analysis systems not connected to the Internet but requiring sub-mgr (e.g. managed by Satellite).
+
+    This test will perform the following operations:
+        - Collect the enabled repositories prior to the analysis start
+        - Run the analysis and assert that we successfully enabled the RHSM repositories
+        - Collect the enabled repositories after the tool run to compare with the repositories prior to the analysis
+    """
+    enabled_repositories_prior_analysis = collect_enabled_repositories(shell)
+
+    with convert2rhel(
+        "analyze -y -k {} -o {} --debug".format(
+            TEST_VARS["SATELLITE_KEY"],
+            TEST_VARS["SATELLITE_ORG"],
+        )
+    ) as c2r:
+        c2r.expect("Rollback: Enabling RHSM repositories")
+
+    assert c2r.exitstatus == 0
+
+    enabled_repositories_after_analysis = collect_enabled_repositories(shell)
+
+    assert enabled_repositories_prior_analysis == enabled_repositories_after_analysis
```

Comment:
After some review of this, I think this scenario cannot work. It does the following:

1. register to satellite (the registration account contains only the original system repos )-> the system has enabled original non rhel repos from satellite
2. call c2r with satellite credentials -> the c2r will try to re-register with the new credentials (the ones containing rhel repos). This means that the system has to unregister first and thus loses access to the original registration
3. after the analysis the system doesn't have the original repositories tight the original pre-registration so the assert fails. The c2r has no idea how to re-register using the original registration (because it doesn't know the credentials to that). 
Now I'm not sure if it is possible to somehow backup the original entitlements and somehow use them after the analysis to re-register but I wouldn't even go this way..
I think the scenario should be (based on the [ticket description](https://issues.redhat.com/browse/RHELC-1495)):

1. register to satellite using the account/credentials that contain both the rhel and original distro repositories
2. run the c2r without passing any credentials
3. assert that the enabled repos on the system are the same as before running the analysis

@danmyway is currently working on getting the satellite account with both OSes repos

Reply:
Removed the credentials from the test. Let's see if that will work

Reply:
@r0x0d ok let me just run the test using the tesar

Reply:
@kokesak the both system repos activation key is here: https://gitlab.cee.redhat.com/oamg/convert2rhel/convert2rhel-secrets/-/merge_requests/26?commit_id=588d3748d348186cab41fd3a763c5b5b7295cc6b

Reply:
yeah it failed http://artifacts.osci.redhat.com/testing-farm/69a68ab6-bbc2-413e-8773-bccfa534654a/, (it is passed because analysis exits with 0 even if inhibitor raises)

Reply:
@r0x0d I added few comments below to "fix" the integration test. Now it should be doing what it is supposed to, but like Dan said on the standup - the repos on stallite may not be synced :(. Let's see the results